tests/zfs-tests/tests/functional/cli_root/zfs_load-key/zfs_load-key_common.kshlib

```diff
@@ -99,3 +99,62 @@ function verify_origin
 
 	return 0
 }
+
+function setup_https
+{
+	log_must openssl req -x509 -newkey rsa:4096 -sha256 -days 1 -nodes -keyout "/$TESTPOOL/snakeoil.key" -out "$SSL_CA_CERT_FILE" -subj "/CN=$HTTPS_HOSTNAME"
+
+	python3 -uc "
+import http.server, ssl, sys, os, time
+
+sys.stdin.close()
+
+httpd, err = None, None
+for i in range(1, 6):
+	try:
+		httpd = http.server.HTTPServer(('$HTTPS_HOSTNAME', $HTTPS_PORT), http.server.SimpleHTTPRequestHandler)
+		break
+	except:
+		err = sys.exc_info()[1]
+		time.sleep(i / 10)
+if not httpd:
+	raise err
+
+httpd.socket = ssl.wrap_socket(httpd.socket, server_side=True, keyfile='/$TESTPOOL/snakeoil.key', certfile='$SSL_CA_CERT_FILE', ssl_version=ssl.PROTOCOL_TLS)
+
+os.chdir('$STF_SUITE/tests/functional/cli_root/zfs_load-key')
+
+with open('/$TESTPOOL/snakeoil.pid', 'w') as pidf:
+	if os.fork() != 0:
+	  os._exit(0)
+	print(os.getpid(), file=pidf)
+
+sys.stdout.close()
+sys.stderr.close()
+try:
+	sys.stdout = sys.stderr = open('/tmp/snakeoil.log', 'w', buffering=1) # line
+except:
+	sys.stdout = sys.stderr = open('/dev/null', 'w')
+
+print('{} start'.format(os.getpid()))
+httpd.serve_forever()
+" 2>&1 || log_fail
```

Comment:
Should give a message explaining what failed. And why redirect stderr?

Reply:
It does. The stderr redirexion was left over from testing.